src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -110,88 +245,133 @@ dx_path() {
       ;;
   esac
 }
-    
+
 main() {
-    if [[ $debug == true ]]; then
-      export NXF_DEBUG=2
-      TRACE_CMD="-trace nextflow.plugin"
-      set -x && env | grep -v DX_SECURITY_CONTEXT | sort
-    fi
-    
-    if [ -n "$docker_creds" ]; then
-        dx-registry-login
-    fi
+  if [[ $debug == true ]]; then
+    export NXF_DEBUG=2
+    TRACE_CMD="-trace nextflow.plugin"
+    env | grep -v DX_SECURITY_CONTEXT | sort
+    set -x
+  fi
 
-    DX_WORK=${work_dir:-$DX_WORKSPACE_ID:/scratch/}
-
-    LOG_NAME="nextflow-$(date +"%y%m%d-%H%M%S").log"
-    # parse dnanexus-job.json to get job output destination
-    OUT_PROJECT=$(jq -r .project /home/dnanexus/dnanexus-job.json)
-    OUT_FOLDER=$(jq -r .folder /home/dnanexus/dnanexus-job.json)
-    OUTDIR="$OUT_PROJECT:${OUT_FOLDER#/}"
-    DX_LOG=${log_file:-"$OUTDIR/$LOG_NAME"}
-
-    export NXF_WORK=dx://$DX_WORK
-    export NXF_HOME=/opt/nextflow
-    export NXF_UUID=$(uuidgen)
-    export NXF_ANSI_LOG=false
-    export NXF_EXECUTOR=dnanexus
-    export NXF_PLUGINS_DEFAULT=nextaur@1.1.0
-    export NXF_DOCKER_LEGACY=true
-    #export NXF_DOCKER_CREDS_FILE=$docker_creds_file
-    #[[ $scm_file ]] && export NXF_SCM_FILE=$(dx_path $scm_file 'Nextflow CSM file')
-
-    mkdir -p /home/dnanexus/out/output_files
-    cd /home/dnanexus/out/output_files
-    required_inputs=""
-    @@REQUIRED_RUNTIME_PARAMS@@
+  DX_CACHEDIR=$DX_PROJECT_CONTEXT_ID:/.nextflow_cache_db
+  validate_run_opts
+
+  if [[ $preserve_cache == true ]]; then
+    check_cache_db_storage
+  fi
+
+  if [ -n "$docker_creds" ]; then
+    dx-registry-login
+  fi
+
+  # set default NXF env constants
+  export NXF_DOCKER_LEGACY=true
+  #export NXF_DOCKER_CREDS_FILE=$docker_creds_file
+  #[[ $scm_file ]] && export NXF_SCM_FILE=$(dx_path $scm_file 'Nextflow CSM file')
+  export NXF_HOME=/opt/nextflow
+  export NXF_ANSI_LOG=false
+  export NXF_PLUGINS_DEFAULT=nextaur@1.1.0
```

Comment:
Version of nextaur could be set at the top (it would be easier to change it in case of nextaur version change.)

Reply:
Done